app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorEffectHandler.kt

```diff
@@ -59,4 +63,13 @@ class SyncIndicatorEffectHandler @AssistedInject constructor(
           }
     }
   }
+
+  private fun startDataSync(): ObservableTransformer<InitiateDataSync, SyncIndicatorEvent> {
+    return ObservableTransformer { effect ->
+      effect
+          .doOnNext { dataSync.fireAndForgetSync(SyncGroup.FREQUENT) }
+          .switchMap { dataSync.streamSyncErrors() }
```

Comment:
Not necessary to change this now, but this need not be a part of this stream, right? What we want is to be notified whenever there is a sync error, in which case this can be set up as a separate effect which continuously observes the stream for errors.

Reply:
Never mind, this is intentional because we show a pop-up and we want it to be shown only if the user the manually triggered a sync.

Should we add a comment here explaining this?

Reply:
I will add a comment. Also the naming below is misleading.  `dataSync.fireAndForgetSync(SyncGroup.FREQUENT)`

Reply:
This was meant to expose the detail that if you call this method, the sync is going to get triggered and no more action is required. Would you mind explaining what you find misleading about this? We can think of alternate names as well.

Reply:
Sorry, I meant misleading because in this context it wasn't a `fireAndForgetSync` call. We are listening to errors here after we initiate sync. But I think when the sync call itself was moved to separate function, this became a fire and forget.

Reply:
Right, but the listening to sync errors is a separate call since you can invoke `fireAndForgetSync` and not have to subscribe to the errors at all.

But I see what you mean and I think a root cause here is that the api of the entire sync layer and how it exposes information about ongoing syncs and triggering them is poorly designed.